scripts/profiler/model_creator.py

```diff
@@ -23,9 +23,10 @@ class Command(Enum):
 
 class ModelCreator:
 
-    def __init__(self, own_recv_socket_dict,
-                 own_send_socket_dict=None,
-                 remote_socket_dict=None,
+    def __init__(self, pipe_recv_desc,
```

Comment:
Maybe we could actually provide streams to the Python modules. Thanks to that modules implementation would be independent to way we create streams (and we can assume that basic stream API would not change no matter how the stream is implemented).